GitUpKit/Third-Party/Package.swift

```diff
@@ -0,0 +1,211 @@
+// swift-tools-version:5.3
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+import Foundation
+
+let libgit2OriginPath = "./libgit2"
+let httpClientPath = "\(libgit2OriginPath)/deps/http-parser"
+let ntlmClientPath = "\(libgit2OriginPath)/deps/ntlmclient"
+
+let librariesPath = "."
+let libssh2Path = "\(librariesPath)/libssh2.xcframework"
+let libsslPath = "\(librariesPath)/libssl.xcframework"
+let libcryptoPath = "\(librariesPath)/libcrypto.xcframework"
+
+enum FeaturesExtractor {
+    private struct Define: CustomStringConvertible {
+        let define: String
+        let value: String
+        var description: String {
+            "\(define) \(value)"
+        }
+    }
+    private static var packageSwiftDirectory: URL? {
+        var directory = URL.init(fileURLWithPath: "\(#file)")
+        if directory.lastPathComponent == "Package.swift" {
+            directory.deleteLastPathComponent()
+        }
+        return directory
+    }
+    
+    private static var libgit2Directory: URL? {
+        packageSwiftDirectory?.appendingPathComponent("libgit2")
+    }
+    
+    private static var featuresPath: URL {
+        libgit2Directory?.appendingPathComponent("include/git2/sys/features.h") ?? .init(fileURLWithPath: "")
+    }
+    
+    private static func shouldAddExtraDefine(featuresPath: URL, define: Define) -> Bool {
+        
+        guard let string = try? String.init(contentsOf: featuresPath, encoding: .utf8) else {
+            return false
+        }
+        
+        return !string.contains("#define \(define)")
+    }
+    
+    private static func fixedExtraDefine(featuresPath: URL, define: Define) -> [CSetting] {
+        shouldAddExtraDefine(featuresPath: featuresPath, define: define) ? [ .define(define.define, to: define.value) ] : []
+    }
+    
+    private static func fixedExtraDefine(define: Define) -> [CSetting] { fixedExtraDefine(featuresPath: featuresPath, define: define) }
+
+    private static func fixedExtraSSHDefines() -> [CSetting] {
+        let defines: [Define] = [
+            .init(define: "GIT_SSH", value: "1"),
+            .init(define: "GIT_SSH_MEMORY_CREDENTIALS", value: "1")
+        ]
+        return defines.flatMap(fixedExtraDefine(define:))
+    }
+    
+    static func extraLibgit2CSettings() -> [CSetting] {
+        let fixedLibgit2CSettings = fixedExtraSSHDefines()
+        return fixedLibgit2CSettings
+    }
+}
+
+let package = Package(
+    name: "SwiftPackage",
```

Comment:
@lolgear I found the source of my warnings with futimens. Turns out that swift packages defaults to 10.10 for its minimum target. I confirmed this with the build logs `-target arm64-apple-macos10.10`. After making the change I'm suggesting, it builds with `-target arm64-apple-macos10.13`. To bump this we need to define the following, let's but it below `name`:

```
platforms: [
    .macOS(.v10_13)
],
```

Reply:
Well, all warnings are coming from broken CMake. If you use CMake 2.22, it will take into account environment variable in script and correctly cut out futimens for libgit2.
But recent version of CMake uses a different mechanism and ignores current environment variable.

Reply:
Well I don't want to rely on a particular version cmake. Let's define the platform and we'll always get what we want. I've already bumped to 10.13.

Reply:
Also...I don't think the warnings (for this issue) are on CMake's end for me, it seems like it compiles correctly with that value looking at the CMake output logs. It's when compiling the `Libgit2Origin` target in Xcode that this warning surfaces:

```
CompileC /Users/lucasderraugh/Library/Developer/Xcode/DerivedData/GitUp-bqeuvsqkhipodmffbhvhltdikopc/Build/Intermediates.noindex/SwiftPackage.build/Debug/Libgit2Origin.build/Objects-normal-asan/arm64/xprepare.o /Users/lucasderraugh/Developer/GitUp/GitUpKit/Third-Party/libgit2/src/xdiff/xprepare.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Libgit2Origin' from project 'SwiftPackage')
    cd /Users/lucasderraugh/Developer/GitUp/GitUpKit/Third-Party
    export LANG\=en_US.US-ASCII
    /Applications/Xcode-13.4.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -target arm64-apple-macos10.10
```

Reply:
It works with previous CMake and libgit2 maintainers added futiments function to search for futimens function, because CMake can't find it.
It works for previous CMake and now it doesn't :)
CMake decided to change something and here we are :)
I'll update both constant in script and platform version.

Reply:
Thank you I greatly appreciate your work on this! I'm out tomorrow but will try to merge this in as long as the tests still work on the Mac side (the iOS ones I think are already broken though I should probably fix them)